config.c

```diff
@@ -0,0 +1,302 @@
+ConfigItems configItems; 
+
+BOOL CreateDefaultConfigFile(HINSTANCE);
+BOOL LoadDefaultConfigResourceData(HINSTANCE);
+BOOL WriteDefaultConfigDataToFile(); 
+void trim(char* str); 
+void removeControlChars(char* str); 
+size_t GetLineCount(FILE* file);
+ 
+DWORD ReadConfigFile() 
+{ 
+	//Try to open the config file based on the path
+	FILE* configFileHandle = _wfopen(*szConfigFilePath, L"r"); 
+	
+	if(configFileHandle == NULL) 
+	{
+		reportWin32Error(L"Config file could not be opened"); 
+		CleanupConfigReader();
+		return ERROR_INVALID_HANDLE; 
+	}
+	
+	char line[BUFF_SIZE]; //TODO must have a more clever way of getting a line length
+		
+	configItems.configItem = (ConfigItem*)malloc(sizeof(ConfigItem) * GetLineCount(configFileHandle) + 1); 
+	configItems.configItemsCount = GetLineCount(configFileHandle) + 1; 
+
+	if(configItems.configItem == NULL) 
+	{ 
+		reportWin32Error(L"Allocation ConfigItem memory"); 
+		CleanupConfigReader(); 
+		return ERROR_NOT_ENOUGH_MEMORY;
+	}
+	
+	for(size_t lineCount = 0; fgets(line, sizeof(line), configFileHandle); lineCount++) 
+	{ 
+		if(strlen(line) == 0)
+		{
+			continue; 
+		}
+		
+		//Get the first half of the line
+		char* token = strtok(line, " "); 
+		configItems.configItem[lineCount].name = (char*)malloc(strlen(token) + 1);
+		strncpy(configItems.configItem[lineCount].name, token, strlen(token) + 1); 
+		
+		//Get the second half of the line
+		token = strtok(NULL, " "); 
+		removeControlChars(token);
+		configItems.configItem[lineCount].value = (char*)malloc(strlen(token) + 1); 
+		strncpy(configItems.configItem[lineCount].value, token, strlen(token) + 1);
+		
+		DEBUG_PRINT("Name: %s Value: %s Name LEN: %lu Value LEN: %lu Count: %lu", 
+			configItems.configItem[lineCount].name, 
+			configItems.configItem[lineCount].value,
+			strlen(configItems.configItem[lineCount].name), 
+			strlen(configItems.configItem[lineCount].value), 
+			lineCount); 
+	}
+	
+	
+	fclose(configFileHandle); 
+	
+	return ERROR_SUCCESS;
+}
+
+void GetConfigFilePath() 
+{ 
+	//TODO: We don't check other results possible results, i.e E_FAIL
+	HRESULT getAppDataPathResult = SHGetKnownFolderPath(&FOLDERID_RoamingAppData, 0, NULL, szConfigFilePath); 
+	
+	if(!SUCCEEDED(getAppDataPathResult)) { 
+		reportWin32Error(L"Could not get the users appdata directory"); 
+		CoTaskMemFree(szConfigFilePath); 
+		exit(ERROR_PATH_NOT_FOUND);
+	}
+	
+	HRESULT concatStringResult = StringCchCatW(*szConfigFilePath, MAX_PATH, L"\\lightwm.config");
+	
+	if(!SUCCEEDED(concatStringResult)) { 
+		switch(concatStringResult){ 
+			case STRSAFE_E_INVALID_PARAMETER: 
+				SetLastError(ERROR_INVALID_PARAMETER); 
+				break; 
+			case STRSAFE_E_INSUFFICIENT_BUFFER: 
+				SetLastError(ERROR_INSUFFICIENT_BUFFER);
+				break; 
+		}
+		reportWin32Error(L"Could not append file name to appdata path"); 
+		CoTaskMemFree(szConfigFilePath); 
+		exit(GetLastError()); 
+	}
+}
+
+uint8_t LoadConfigFile(HINSTANCE resourceModuleHandle) 
```

Comment:
As a convention I prefer the non-windows api functions to be camelCase to avoid confusion